lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -148,11 +149,51 @@ export default function getAPIImplementation(): APIInterface {
             }
 
             // we find the primary user ID from the user's list for later use.
-            let primaryUserAssociatedWithEmail = users.find((u) => u.isPrimaryUser);
+            let linkingCandidate = users.find((u) => u.isPrimaryUser);
 
-            // first we check if there even exists a primary user that has the input email
+            logDebugMessage("generatePasswordResetTokenPOST: primary linking candidate: " + linkingCandidate?.id);
+
+            // If there is no existing primary user and there is a single option to link
+            // we see if that user can become primary (and a candidate for linking)
+            if (linkingCandidate === undefined && users.length === 1) {
+                // If the only user that exists with this email is a non-primary emailpassword user, then we can just let them reset their password, because:
```

Comment:
Add a few test cases:
- SSO login with email A, email password login with email A (they are linked), change ep login to email B -> password reset generation should be denied for email B.
- SSO login with email A (unverified, recipe user). Reset password flow with email A -> should generate token -> but during consuming token, and not link to SSO user.
- We have SSO user with email A, not verified (not primary or primary, doesn't matter). Then we try and go through password reset flow with email A. The shouldDoAutomaticAccountLinking definition is https://gist.github.com/rishabhpoddar/7948a544acfb66b26f60d31c113074c9. So this should not allow reset password.
- We have SSO user with email A, verified (not primary or primary, doesn't matter). Then we try and go through password reset flow with email A. The shouldDoAutomaticAccountLinking definition is https://gist.github.com/rishabhpoddar/7948a544acfb66b26f60d31c113074c9. So this case, it should do password reset, and it should link to the existing SSO user.
- We have SSO user with email A, not verified (not primary), email password user exists (not primary or primary, doesn't matter). Then we try and go through password reset flow with email A. The shouldDoAutomaticAccountLinking definition is https://gist.github.com/rishabhpoddar/7948a544acfb66b26f60d31c113074c9. So this should allow password reset, but it should not link it.